lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
To be frank, this just looks like overly complex java code. There's no reason for class factories and they just provide complexity that I'm having a hard time seeing the need for.

e.g. how is that any better than $client = new Client(...)->setX();